app/src/main/java/com/base/firebase_operations/MainActivity2.kt

```diff
@@ -0,0 +1,66 @@
+package com.base.firebase_operations
+
+import android.content.Intent
+import android.net.Uri
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class MainActivity2 : AppCompatActivity() {
+
+    private val Requestcode=0
+    private var currentfile: Uri?=null
+    lateinit var imageholder:ImageView
+    val imagereff=Firebase.storage.reference
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main2)
+        imageholder=findViewById(R.id.viewimage)
+        val upload: Button=findViewById(R.id.button5)
+        imageholder.setOnClickListener{
+            Intent(Intent.ACTION_GET_CONTENT).also {
+                it.type="image/*"
+                startActivityForResult(it,Requestcode)
+            }
+        }
+        upload.setOnClickListener {
+            UploadImagetoCloud("TestImages")
+        }
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if(resultCode== RESULT_OK && requestCode==Requestcode){
+            data?.data?.let{
+                currentfile=it
+                imageholder.setImageURI(it)
+            }
+        }
+    }
+
+    private fun UploadImagetoCloud(filename:String)= CoroutineScope(Dispatchers.IO).launch {
```

Comment:
So now we just created a function which would start a coroutine on the IO thread and perform the image uploading task.
So the function just takes a filename which is the name of file which would be stored at the Cloud storage.
Then we just checked if the currentfile is not null and if so used a let block and just used the cloud storage reference and passed the file name and child name to it and used the putfile method which would take the image uri and save it to the Cloud storage.